lisa/messages.py

```diff
@@ -314,6 +315,10 @@ def create_test_result_message(
     message.status = test_status
     message.message = test_message
     message.elapsed = test_result.get_elapsed()
+    if message_type == SubTestMessage:
+        if not other_fields:
+            other_fields = {}
+        other_fields.update({"parent_test": test_result.runtime_data.name})
```

Comment:
it should be filled in 
other_fields["information"]["parent_test"]

this is current data run from your changed code
```
2023-04-12 14:25:32.872[5320][DEBUG] lisa.notifier[Console] received message [SubTestResult]: SubTestMessage(type='SubTestResult', time=datetime.datetime(2023, 4, 12, 14, 25, 32, 794499), elapsed=2972.2109429, id_='lisa_0_0', name='binfmt_misc02', status=<TestStatus.PASSED: 5>, message='', stacktrace=None, information={'version': '20200930', 'exit_value': 0}, hardware_platform='x86_64', parent_test='ltp_lite')
```

Reply:
I initially used 'information' field to store 'parent_test' but Chi suggested to add a field in SubTestMessage itself instead. And I agree with that idea as well.

Reply:
It's a common requirement to know the parent test name, so I asked to add it to subtest message level.